web/src/main/java/com/microsoft/applicationinsights/web/internal/correlation/tracecontext/Traceparent.java

```diff
@@ -0,0 +1,189 @@
+     * @param spanId
+     * @param traceFlags
+     */
+    public Traceparent(int version, String traceId, String spanId, int traceFlags) {
+        this(version, traceId != null ? traceId : randomHex(16),
+            spanId != null ? spanId : randomHex(8),
+            traceFlags, true);
+    }
+
+    /**
+     * This constructor creates a new Traceparent object having new traceId.
+     * It should only be used if the call is the starting point of distributed trace.
+     */
+    public Traceparent() {
+        this(0, randomHex(16), randomHex(8), 0, false);
+    }
+
+    public String getTraceId() {
+        return traceId;
+    }
+
+    public int getTraceFlags() {
+        return traceFlags;
+    }
+
+    public String getSpanId() {
+        return spanId;
+    }
+
+    /**
+     * Validates the given input based on W3C specifications.
+     * @param version
+     * @param traceId
+     * @param spanId
+     * @param traceFlags
+     * @throws IllegalArgumentException
+     */
+    private static void validate(int version, String traceId, String spanId, int traceFlags)
+        throws IllegalArgumentException {
+        if (version < 0 || version > 254) {
+            throw new IllegalArgumentException("version must be within range [0, 255)");
+        }
+        if (!isHex(traceId, 32)) {
+            throw new IllegalArgumentException("invalid traceId");
+        }
+        if (traceId.equals("00000000000000000000000000000000")) {
+            throw new IllegalArgumentException("invalid traceId");
+        }
+        if (!isHex(spanId, 16)) {
+            throw new IllegalArgumentException("invalid spanId");
+        }
+        if (spanId.equals("0000000000000000")) {
+            throw new IllegalArgumentException("invalid spanId");
+        }
+        if (traceFlags < 0 || traceFlags > 255) {
+            throw new IllegalArgumentException("traceFlags must be within range [0, 255]");
+        }
+    }
+
+    /**
+     * Converts the Traceparent object to header format
+     * Eg: 00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-01
+     * @return traceparent
+     */
+    @Override
+    public String toString() {
+        return String.format("%02x-%s-%s-%02x", version, traceId, spanId, traceFlags);
+    }
+
+    /**
+     * Helper method to create a random hexadecimal string of n bytes.
+     * @param n
+     * @return n byte hexadecimal string
+     */
+    @VisibleForTesting static String randomHex(int n) {
+        byte[] bytes = new byte[n];
+        ThreadLocalRandom.current().nextBytes(bytes);
+        StringBuilder sb = new StringBuilder();
+        for (byte b : bytes) {
+            sb.append(String.format("%02x", b));
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Helper method to check if a given string of n bytes is hexadecimal
+     * @param s
+     * @param n
+     * @return boolean
+     */
+    private static boolean isHex(String s, int n) {
+        if (s == null || s.length() == 0) {
```

Comment:
One option is to `Long.parseLong(s, 16)` and if it throws NumberFormatException, return false.

Sometimes try/catch can be more performant, but if that's not a consideration at this point, I wouldn't worry about it. Also, I'd write a test to confirm prove it's more performant if you do choose that method.